packages/frontend/src/components/ChartConfigPanel/Legend/ReferenceLines.tsx

```diff
@@ -0,0 +1,154 @@
+        }, [dirtyEchartsConfig?.series]);
+
+    const [value, setValue] = useState<string | undefined>(
+        selectedMarklineValue,
+    );
+
+    const selectedFieldDefault = useMemo(() => {
+        if (selectedMarklineAxis === undefined) return;
+        return fieldsInAxes.find((field) => {
+            const fieldId = isField(field) ? getFieldId(field) : field.name;
+            return fieldId === selectedFieldId;
+        });
+    }, [fieldsInAxes]);
+
+    const [selectedField, setSelectedField] = useState<
+        Field | TableCalculation | CompiledDimension | undefined
+    >(selectedFieldDefault);
+
+    const [isOpen, setIsOpen] = useState<boolean>(
+        selectedMarklineAxis !== undefined,
+    );
+
+    useEffect(() => {
+        if (value && selectedField) {
+            const fieldId = isField(selectedField)
+                ? getFieldId(selectedField)
+                : selectedField.name;
+
+            if (dirtyEchartsConfig?.series) {
+                let alreadyAdded = false; // flag to avoid duplication on Xaxis
+                const series = dirtyEchartsConfig?.series.map((serie) => {
+                    const axisRef =
+                        dirtyLayout?.xField === fieldId
+                            ? serie.encode.xRef
+                            : serie.encode.yRef;
+                    if (axisRef.field === fieldId && !alreadyAdded) {
+                        alreadyAdded = true;
+                        const axis =
+                            dirtyLayout?.xField === fieldId ? 'xAxis' : 'yAxis';
+                        return {
+                            ...serie,
+                            markLine: { data: [{ [axis]: value }] },
+                        };
+                    } else {
+                        // Remove markLine for the rest of series
+                        return {
+                            ...serie,
+                            markLine: undefined,
+                        };
+                    }
+                });
+                updateSeries(series);
+            }
+        }
+    }, [value, selectedField, updateSeries]);
+
+    return (
+        <>
+            <Checkbox
+                name="show"
+                onChange={() => setIsOpen(!isOpen)}
+                checked={isOpen}
+            >
+                Include reference line
+            </Checkbox>
+
+            <Collapse isOpen={isOpen}>
+                <SectionTitle>Line 1</SectionTitle>
+                <GridSettings>
+                    <Label>Field</Label>
+                    <FieldAutoComplete
+                        fields={fieldsInAxes}
+                        activeField={selectedField}
+                        onChange={(item) => {
+                            setSelectedField(item);
+                        }}
+                    />
+                </GridSettings>
+                <GridSettings>
+                    <Label>Value</Label>
+
+                    <InputGroup
```

Comment:
The reason why I'm using an InputGroup isntead of a NumericInput is because I want to type `decimal` numbers (for %) not just integers.